presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -0,0 +1,1970 @@
+            if (lastThrowableFromIrreversibleOperations != null) {
+                throw new PrestoException(
+                        HIVE_METASTORE_ERROR,
+                        format(
+                                "Failed to execute all metastore delete operations: %s",
+                                failedIrreversibleOperationDescriptions.stream()
+                                        .collect(Collectors.joining("; "))),
+                        lastThrowableFromIrreversibleOperations);
+            }
+
+            // If control flow reached this point, this commit is considered successful no matter
+            // what happens later. The only kind of operations that haven't been carried out yet
+            // are cleanups.
+
+            // The program control flow will go to finally next. And cleanup will run because
+            // moveForwardInFinally has been set to false.
+        }
+        finally {
+            if (moveForwardInFinally) {
+                // After this line, all operations are best-effort clean up operations.
+                // If any operation fails, the error will be logged and ignored.
+                // Additionally, other clean up operations should still be attempted.
+                List<DirectoryCleanUpTask> forwardCleanupTask = null;
+                List<DeclaredIntentionToWrite> stagedIntentionToWrite = null;
+                try {
+                    // Compute values needed for failure logging first
+                    forwardCleanupTask = cleanUpTasks.stream()
+                            .filter(task -> task.getKind() == DirectoryTaskKind.MOVE_FORWARD)
+                            .collect(Collectors.toList());
+                    stagedIntentionToWrite = declaredIntentionsToWrite.stream()
+                            .filter(intention -> intention.getMode() == WriteMode.STAGE_AND_MOVE_TO_TARGET_DIRECTORY)
+                            .collect(Collectors.toList());
+
+                    // Execute deletion tasks
+                    for (DirectoryCleanUpTask cleanUpTask : forwardCleanupTask) {
+                        verify(cleanUpTask.isDeleteEmptyDirectory());
+                        if (!deleteRecursivelyIfExists(cleanUpTask.getUser(), hdfsEnvironment, cleanUpTask.getPath())) {
+                            logCleanupFailure("Error deleting directory %s", cleanUpTask.getPath().toString());
+                        }
+                    }
+
+                    // Clean up empty staging directories (that may recursively contain empty directories)
+                    for (DeclaredIntentionToWrite declaredIntentionToWrite : stagedIntentionToWrite) {
+                        Path path = declaredIntentionToWrite.getRootPath();
+                        // Should be ok to just rmr. @dain, opinion?
+                        RecursiveDeleteResult recursiveDeleteResult = recursiveDeleteFiles(hdfsEnvironment, declaredIntentionToWrite.getUser(), path, ImmutableList.of(), true);
```

Comment:
I don't think this will delete anything because the prefix list is empty.  Is there any reason to not use our query prefix here?